core/os/android/adb/device.go

```diff
@@ -110,6 +111,17 @@ func Devices(ctx context.Context) (DeviceList, error) {
 	return out, nil
 }
 
+func gapidApkPackageName(abi *device.ABI) string {
+	switch {
+	case abi.SameAs(device.AndroidARMv7a):
+		return "com.google.android.gapid.armeabiv7a"
```

Comment:
Did we package the 64-bit binary in the gapid-armeabi-v7a.apk?  It's good if we did, I just want to check.

What about on x86?  In that case, the launcher won't run.

Reply:
No, we compile a 32-bit binary version of the `launch_producer` in that case which launches 32-bits producer if available. Not sure whether we pack the `launch_producer` for x86 but if it can't find the producer for that ABI it fails.

Reply:
For x86, it's fine to just fail (assume we have proper error message of course)
For 32-bits, I'm questioning whether packaging the binary into gapid-armeabi-v7a.apk is the right call.  If we do, we should package the same binary.  The counter producer we run should not depend on the app we are profiling.  We should always run the 64-bit counter producer, because are not requiring the vendors to provide both 32-bits and 64-bits version.
It's fine to get something working first.  But please do file a new bug to follow up on this if it isn't resolve in this PR.

Reply:
GAPID pushes the apk based on the primary ABI of the device. If the device is 32-bits and there's no 32-bits producer provided, then it will fail anyway.